include/deal.II/base/memory_space_data.h

```diff
@@ -44,6 +156,36 @@ namespace MemorySpace
                     "MemorySpace should be Host or CUDA");
     }
 
+    bool
+    has_data()
+    {
+      return false;
```

Comment:
This seems odd. Maybe I don't understand what the function refers to.

Reply:
Oh, I get it now. This is just a dummy implementation. I would prefer it if you moved the class declaration out of the `#ifdef` and instead only guard the relevant member variables plus the *implementation* of member functions (moved out of line) with the `#ifdef`.